src/Avalonia.Controls/DateTimePickers/DatePicker.cs

```diff
@@ -399,9 +399,9 @@ private void SetSelectedDateText()
             else
             {
                 PseudoClasses.Set(":hasnodate", true);
-                _monthText!.Text = "month";
-                _yearText!.Text = "year";
-                _dayText!.Text = "day";
+                _monthText!.Text ??= "month";
+                _yearText!.Text ??= "year";
+                _dayText!.Text ??= "day";
```

Comment:
This will fail if you clear the selected date (or time) after having it previously set since the TextBlock text won't be null. WinUI probably has localized strings here to prevent English only. 
`TextBlock.Text` is also a `DirectProperty`, not a `StyledProperty`, so there's a chance the Style lookup in the OP will fail, and may not even work here to allow custom strings for `:hasnodate` (or `:hasnotime`)

Reply:
Yes, property localization for internal avalonia strings is something that we must have. It needs a proper design though.